tree/ntuple/v7/test/ntuple_compat.cxx

```diff
@@ -259,16 +259,78 @@ TEST(RNTupleCompat, FutureColumnType)
    {
       auto model = RNTupleModel::Create();
       auto col = model->MakeField<ROOT::Experimental::Internal::RTestFutureColumn>("futureColumn");
-      auto writeOpts = RNTupleWriteOptions();
-      writeOpts.SetCompression(0);
-      auto writer = RNTupleWriter::Recreate(std::move(model), "ntpl", fileGuard.GetPath(), writeOpts);
+      auto colValid = model->MakeField<float>("float");
+      auto writer = RNTupleWriter::Recreate(std::move(model), "ntpl", fileGuard.GetPath());
       col->dummy = 0x42424242;
+      *colValid = 69.f;
       writer->Fill();
    }
 
    auto reader = RNTupleReader::Open("ntpl", fileGuard.GetPath());
    const auto &desc = reader->GetDescriptor();
    const auto &fdesc = desc.GetFieldDescriptor(desc.FindFieldId("futureColumn"));
+   GTEST_ASSERT_EQ(fdesc.GetLogicalColumnIds().size(), 1);
    const auto &cdesc = desc.GetColumnDescriptor(fdesc.GetLogicalColumnIds()[0]);
    EXPECT_EQ(cdesc.GetType(), EColumnType::kUnknown);
+
+   {
+      // Creating a model not in fwd-compatible mode should fail
+      EXPECT_THROW(desc.CreateModel(), RException);
+   }
+
+   {
+      auto modelOpts = RNTupleDescriptor::RCreateModelOptions();
+      modelOpts.fForwardCompatible = true;
+      const auto &model = desc.CreateModel(modelOpts);
+
+      // The future column should not show up in the model
+      EXPECT_THROW(model->GetField("futureColumn"), RException);
+
+      const auto &floatFld = model->GetField("float");
+      EXPECT_EQ(floatFld.GetTypeName(), "float");
```

Comment:
Maybe also read back the value (pass the model to a reader; we can later add a fwd compat option to the RNTupleReadOptions)